### ISSUE TITLE
Same labels and predictions, non-zero loss

### ISSUE THREAD

**Issue by OverLordGoldDragon** [tensorflow/tensorflow#40025]
```python
import numpy as np
import tensorflow as tf
from tensorflow.keras.layers import Input, Dense, Flatten
from tensorflow.keras.models import Model

#%%#######################################################
ipt = Input(batch_shape=(128, 28, 28, 1))
x   = Flatten()(ipt)
out = Dense(10, activation='softmax')(x)
model = Model(ipt, out)
model.compile('adam', 'categorical_crossentropy')

#%%#######################################################
x = np.random.uniform(0, 1, model.input_shape)
pred = model(x, training=True)  # =False also works
loss = model.compiled_loss(pred, pred)

print(tf.__version__)
print(loss)
```

```python
2.3.0-dev20200531  # Colab; also reproduced in 2.2.0, Win-10
tf.Tensor(1.9904033, shape=(), dtype=float32)
```

What's the deal?

**Comment by OverLordGoldDragon** [comment]
This is expected behavior - my mishap.
